syz-verifier/main.go

```diff
@@ -267,6 +276,70 @@ func (srv *RPCServer) NextExchange(a *rpctype.NextExchangeArgs, r *rpctype.NextE
 	return nil
 }
 
+// newResult is called when a Runner sends a new Result. It returns true if all
+// Results from the corresponding programs have been received and they can be
+// sent for verification. Otherwise, it returns false.
+func (srv *RPCServer) newResult(res *verf.Result, idx int) bool {
+	for _, prog := range srv.progs {
+		if prog.idx == idx {
+			prog.res = append(prog.res, res)
+			delete(prog.left, res.Pool)
+			return len(prog.left) == 0
+		}
+	}
+	return false
+}
+
+// processResults will send a set of complete results for verification and, in
+// case differences are found, it will store a result report highlighting those
+// in th workdir/results directory. If writing the results fails, it returns an
+// error.
+func (vrf *Verifier) processResults(res []*verf.Result, prog *prog.Prog) {
+	rr := verf.Verify(res, prog)
+	if rr == nil {
+		return
+	}
+	data := fmt.Sprintf("Errno mismatches found for program:\n%s\n", rr.Prog)
```

Comment:
Lines 302-316 look like a good candidate for a separate function that turns result into a string or []byte.